src/rust/Dockerfile.Makefile

```diff
@@ -0,0 +1,214 @@
+RUN if test "${SCCACHE_LOCATION}" != "s3" ; then \
+      cd /tmp && \
+      wget -q https://github.com/mozilla/sccache/releases/download/0.2.14/sccache-0.2.14-x86_64-unknown-linux-musl.tar.gz && \
+      tar xvzf sccache-0.2.14-x86_64-unknown-linux-musl.tar.gz && \
+      mkdir -p /home/grapl/bin && \
+      cp sccache-0.2.14-x86_64-unknown-linux-musl/sccache /home/grapl/bin/sccache; \
+    fi
+
+# RUN cd /tmp && \
+#     wget -q https://github.com/mozilla/sccache/releases/download/0.2.14/sccache-0.2.14-x86_64-unknown-linux-musl.tar.gz && \
+#     tar xvzf sccache-0.2.14-x86_64-unknown-linux-musl.tar.gz && \
+#     mkdir /home/grapl/bin && \
+#     cp sccache-0.2.14-x86_64-unknown-linux-musl/sccache /home/grapl/bin/sccache
+
+
+# use docker volume cache for sccache
+FROM sccache AS build-sccache-container
+
+RUN --mount=type=cache,uid=19999,gid=19999,target=/home/grapl/.cargo/registry \
+    --mount=type=cache,uid=19999,gid=19999,target=/home/grapl/sccache \
+    if test "${PROFILE}" = "release"; then \
+      cargo build --target=${TARGET} --release; \
+    elif test "${PROFILE}" = "debug"; then \
+      cargo build --target=${TARGET}; \
+    else \
+      echo "ERROR: Unknown build profile: ${PROFILE}"; \
+      exit 1; \
+    fi
+
+# build-sccache-host is not currently supported
+# FROM sccache AS build-sccache-host
+
+# build using sccache from S3
+FROM sccache AS build-sccache-s3
+
+ARG SCCACHE_BUCKET
+
+ENV SCCACHE_BUCKET=${SCCACHE_BUCKET}
+ENV SCCACHE_S3_KEY_PREFIX=sccache
+
+SHELL ["/bin/bash", "-c"]
+
+RUN --mount=type=cache,uid=19999,gid=19999,target=/home/grapl/.cargo/registry \
+    --mount=type=cache,uid=19999,gid=19999,target=/home/grapl/sccache \
+    --mount=type=secret,uid=19999,gid=19999,id=aws,dst=/home/grapl/awscreds \
+    source /home/grapl/awscreds && \
+    if test "${PROFILE}" = "release"; then \
+      cargo build --target=${TARGET} --release; \
+    elif test "${PROFILE}" = "debug"; then \
+      cargo build --target=${TARGET}; \
+    else \
+      echo "ERROR: Unknown build profile: ${PROFILE}"; \
+      exit 1; \
+    fi
+
+
+# create stage alias for easy reference
+FROM build-sccache-${SCCACHE_LOCATION} AS build
+
+
+# zips
+FROM build AS zips
+
+ENV TAG=latest
+
+SHELL ["/bin/bash", "-c"]
+
+RUN mkdir /home/grapl/zips; \
+    grapl-zip() { \
+      TMPDIR=$(mktemp -d); \
+      cd $TMPDIR; \
+      cp "/home/grapl/target/${TARGET}/${PROFILE}/$1" bootstrap && \
+      zip -q -9 -dg /home/grapl/zips/${f}.zip bootstrap; \
+    }; \
+    for f in analyzer-dispatcher \
+			       generic-subgraph-generator \
+			       graph-merger \
+			       metric-forwarder \
+			       node-identifier \
+			       node-identifier-retry-handler \
+			       sysmon-subgraph-generator \
+			       osquery-subgraph-generator; \
+    do \
+        grapl-zip "$f" & \
+    done; \
+    wait
+
+CMD for f in $(ls /home/grapl/zips | sed -e 's/.zip$//g'); do \
+      cp -r /home/grapl/zips/${f}.zip ./${f}-${TAG}.zip; \
+    done
```

Comment:
Might be nice to extract this to a script that can be run (and tested) independently. I worry about burying significant functionality among one of many build stages in a large Dockerfile. `RUN` and `CMD` directives aren't really suited for inlining complex scripts like this.